doubleml/double_ml_selection.py

```diff
@@ -0,0 +1,607 @@
+            for i_fold in range(self.n_folds):
+                train_inds = smpls[i_fold][0]
+                test_inds = smpls[i_fold][1]
+
+                train_inds_1, train_inds_2 = train_test_split(
+                    train_inds, test_size=0.5, random_state=43, stratify=strata[train_inds]
+                )
+
+                s_train_1 = s[train_inds_1]
+                dx_train_1 = dx[train_inds_1, :]
+
+                ml_pi_prelim = clone(self._learner['ml_pi'])
+                ml_pi_prelim.fit(dx_train_1, s_train_1)
+                pi_hat['preds'] = _predict_zero_one_propensity(ml_pi_prelim, dx)
+                pi_hat['targets'] = s
+
+                pi_hat_d0['preds'] = pi_hat['preds'][test_inds]
+                pi_hat_d0['targets'] = s[test_inds]
+
+                xpi = np.column_stack((x, pi_hat['preds']))
+
+                xpi_train_2 = xpi[train_inds_2, :]
+                d_train_2 = d[train_inds_2]
+                xpi_test = xpi[test_inds, :]
+
+                ml_p_d0 = clone(self._learner['ml_p'])
+                ml_p_d0.fit(xpi_train_2, d_train_2)
+
+                p_hat_d0['preds'] = _predict_zero_one_propensity(ml_p_d0, xpi_test)
+                p_hat_d0['preds'] = 1 - p_hat_d0['preds']
+                p_hat_d0['targets'] = d[test_inds]
+
+                s1_d0_train_2_indices = np.intersect1d(np.where(d == 0)[0],
+                                                       np.intersect1d(np.where(s == 1)[0], train_inds_2))
+                xpi_s1_d0_train_2 = xpi[s1_d0_train_2_indices, :]
+                y_s1_d0_train_2 = y[s1_d0_train_2_indices]
+
+                ml_mu_d0_prelim = clone(self._learner['ml_mu'])
+                ml_mu_d0_prelim.fit(xpi_s1_d0_train_2, y_s1_d0_train_2)
+
+                mu_hat_d0['preds'] = ml_mu_d0_prelim.predict(xpi_test)
+                mu_hat_d0['targets'] = y[test_inds]
+
+                dcontrol.append((d == 0)[test_inds])
+                s_0.append(s[test_inds])
+                y_0.append(y[test_inds])
+
+                mu_d0['preds'].append(mu_hat_d0['preds'])
+                pi_d0['preds'].append(pi_hat_d0['preds'])
+                p_d0['preds'].append(p_hat_d0['preds'])
+                mu_d0['targets'].append(mu_hat_d0['targets'])
+                pi_d0['targets'].append(pi_hat_d0['targets'])
+                p_d0['targets'].append(p_hat_d0['targets'])
+
+            mu_hat_d0['preds'] = np.hstack(mu_d0['preds'])
+            pi_hat_d0['preds'] = np.hstack(pi_d0['preds'])
+            p_hat_d0['preds'] = np.hstack(p_d0['preds'])
+            mu_hat_d0['targets'] = np.hstack(mu_d0['targets'])
+            pi_hat_d0['targets'] = np.hstack(pi_d0['targets'])
+            p_hat_d0['targets'] = np.hstack(p_d0['targets'])
+            s_0 = np.hstack(s_0)
+            y_0 = np.hstack(y_0)
+            dcontrol = np.hstack(dcontrol)
+
+        p_hat_d0['preds'] = _trimm(p_hat_d0['preds'], self._trimming_rule, self._trimming_threshold)
```

Comment:
Trimming is done a bit differently in the sample selection paper (see Appendix E in supplementary materials: https://www.tandfonline.com/doi/suppl/10.1080/07350015.2023.2271071?scroll=top). 
It does not truncate the values but rather drops observations for which the product of the treatment propensity score and the selection propensity score is lower than given threshold. Since such option is not supported by DoubleML, I used the truncating option for now, but I guess I could apply this rule to the product of `pi_hat` and `p_hat` instead of just `p_hat`. 

Moreover, the original R implementation `causalweight` has the option to calculate ATE just for the selected subpopulation and in that case, only `p_hat` is trimmed, but I have not implemented this option yet.

Reply:
But generally the clipping/truncating of values would also be an option right?

Reply:
Yes, it should be fine to use it.